quipucords/tests/api/reports/test_reports.py

```diff
@@ -298,25 +90,227 @@ def deployment_with_logs(settings, faker):
+
+    files_contents = extract_tarball_from_response(tarball_response)
+    assert DEPLOYMENTS_JSON_FILENAME in files_contents
+    tarball_deployments_json = json.loads(
+        files_contents[DEPLOYMENTS_JSON_FILENAME].decode()
+    )
+
+    # Compare tarball details.json contents with our model serialized.
+    model_data_as_json = build_cached_json_report(deployments_report)
+    assert model_data_as_json == tarball_deployments_json
+
+    # Compare tarball deployments.json contents with the standalone API's response.
+    api_deployments_response = django_client.get(
+        DEPLOYMENTS_API_PATH.format(deployments_report.report.id),
+        headers={"Accept": "application/json"},
+    )
+    assert api_deployments_response.ok, api_deployments_response.text
+    api_deployments_json = api_deployments_response.json()
+    assert tarball_deployments_json == api_deployments_json
+
+    # Sanity-check the basic structure of the tarball deployments.json directly.
+    assert tarball_deployments_json["report_id"] == deployments_report.report.id
+    assert tarball_deployments_json["report_type"] == REPORT_TYPE_DEPLOYMENT
+
+
+def test_report_details_csv(django_client, deployment_with_logs: DeploymentsReport):
+    """Test report tarball contains expected details.csv."""
+    deployments_report = deployment_with_logs
+    tarball_response = django_client.get(
+        REPORTS_API_PATH.format(deployments_report.report.id)
+    )
+    assert tarball_response.ok, tarball_response.text
+
+    files_contents = extract_tarball_from_response(tarball_response)
+    assert DETAILS_CSV_FILENAME in files_contents
+    tarball_details_csv = files_contents[DETAILS_CSV_FILENAME].decode()
+
+    # Compare tarball details.csv contents with the standalone API's response.
+    details_csv_response = django_client.get(
+        DETAILS_API_PATH.format(deployments_report.report.id),
+        headers={"Accept": "text/csv"},
+    )
+    assert details_csv_response.ok, details_csv_response.text
+    assert details_csv_response.text == tarball_details_csv
+
+    # Compare tarball details.csv contents with our model serialized.
+    serialized_report_data = get_serialized_report_data(deployments_report.report)
+    report_data_csv = create_details_csv(serialized_report_data)
+    assert report_data_csv == tarball_details_csv
+
+    # Sanity-check the basic structure of the tarball details.csv directly.
+    tarball_details_csv_lines = list(csv.reader(tarball_details_csv.splitlines()))
+    expected_csv_first_line = [
+        "Report ID",
+        "Report Type",
+        "Report Version",
+        "Report Platform ID",
+        "Number Sources",
+    ]
+    assert tarball_details_csv_lines[0] == expected_csv_first_line
+    # Why 12? We expect three separate table headers in the file, two section markers,
+    # four blank lines, and at least one line per table. Yes, this CSV is a mess. >:(
+    assert len(tarball_details_csv_lines) >= 12
```

Comment:
I'm wondering if we should look at the csv contents (but not like the original test was looking).

but considering the csv (hopefully) will be discontinued soon ™️ , this is not really an issue.

Reply:
I was thinking hard about where to stop putting effort into testing the CSV (either here or one of the other tests I rewrote), and I actually started to enumerate all the fields at one point, but it quickly felt bloated and excessive knowing that, yes, we plan to discontinue CSV support in the _very near future_. (If we keep saying that, it'll come true, right?)